addons/hr_timesheet/models/hr_timesheet.py

```diff
@@ -127,10 +127,19 @@ def create(self, vals_list):
         for employee in employees:
             user_map[employee.company_id.id][employee.user_id.id] = employee.id
 
+        employee_ids = set()
         for vals in vals_list:
             # compute employee only for timesheet lines, makes no sense for other lines
-            if not vals.get('employee_id') and vals.get('project_id'):
-                vals['employee_id'] = user_map[vals.get('company_id', self.env.company.id)].get(vals.get('user_id', default_user_id), False)
+            if vals.get('project_id'):
+                employee_id = vals.get('employee_id') or user_map[vals.get('company_id', self.env.company.id)].get(vals.get('user_id', default_user_id), False)
+                employee_ids.add(employee_id)
+                if not vals.get('employee_id'):
+                    vals['employee_id'] = employee_id
+
+        employee_ids = list(employee_ids)
+        if any(not employee_id for employee_id in employee_ids)\
+           or self.env['hr.employee'].search_count([('id', 'in', employee_ids), ('active', '=', False)]):
+            raise AccessError("Each timesheet should be linked to an active employee.")
```

Comment:
Wouldn't it be better preventing any check in case employee_ids is empty?
```suggestion
        if employee_ids:
            employee_ids = list(employee_ids)
            if any(not employee_id for employee_id in employee_ids)\
               or self.env['hr.employee'].search_count([('id', 'in', employee_ids), ('active', '=', False)]):
                raise AccessError("Each timesheet should be linked to an active employee.")
```